src/dashboard.ts

```diff
@@ -0,0 +1,180 @@
+import ApexCharts = require('apexcharts');
+
+const pieChart = document.getElementById('pie-chart');
+
+const dataTarget = pieChart.getAttribute('data-target');
+const plantCategory = JSON.parse(dataTarget);
+
+const plantCategoryAmount = plantCategory.map((item: any) => item.amount);
+const plantCategoryName = plantCategory.map((item: any) => item.name);
+
+const generateColor = () => {
+  return '#' + Math.floor(Math.random() * 16777215).toString(16);
+};
+
+const plantVarietyChart = document.getElementById('area-chart');
+const plantVarietyData = plantVarietyChart.getAttribute('data-target');
+const plantVariety = JSON.parse(plantVarietyData);
+
+const counters = document.querySelectorAll(
+  '.counter',
+) as NodeListOf<HTMLElement>;
+const speed = 400;
+
+counters.forEach(counter => {
+  const animate = () => {
+    const value = +counter.getAttribute('amount');
+    const data = +counter.innerText;
+
+    const time = value / speed;
+    if (data < value) {
+      counter.innerText = Math.ceil(data + time).toString();
+      setTimeout(animate, 1);
+    } else {
+      counter.innerText = value.toString();
+    }
+  };
+
+  animate();
+});
+
+const getChartOptions = () => {
+  return {
+    series: plantCategoryAmount,
+    colors: plantCategoryAmount.map(() => generateColor()),
+    chart: {
+      height: 420,
+      width: '100%',
+      type: 'pie',
+    },
+    stroke: {
+      colors: ['white'],
+      lineCap: '',
+    },
+    plotOptions: {
+      pie: {
+        labels: {
+          show: true,
+        },
+        size: '100%',
+        dataLabels: {
+          offset: -25,
+        },
+      },
+    },
+    labels: plantCategoryName,
+    dataLabels: {
+      enabled: true,
+      style: {
+        fontFamily: 'Inter, sans-serif',
+      },
+    },
+    legend: {
+      position: 'bottom',
+      fontFamily: 'Inter, sans-serif',
+    },
+    yaxis: {
+      labels: {
+        formatter: function (value: string) {
+          return value + '%';
+        },
+      },
+    },
+    xaxis: {
+      labels: {
+        formatter: function (value: string) {
+          return value + '%';
+        },
+      },
+      axisTicks: {
+        show: false,
+      },
+      axisBorder: {
+        show: false,
+      },
+    },
+  };
+};
+
+if (document.getElementById('pie-chart') && typeof ApexCharts !== 'undefined') {
```

Comment:
May be we check if ApexCharts is not undefined and do early return if it is undefined?